autopilot/contractor.go

```diff
@@ -177,8 +183,24 @@ func (c *contractor) performContractMaintenance(ctx context.Context, w Worker) (
 		c.logger.Warn("could not calculate min score, no hosts found")
 	}
 
+	// prepare hosts for cache
+	f := newIPFilter(c.logger)
+	gc := worker.NewGougingChecker(state.gs, state.rs, state.cs, state.fee, state.cfg.Contracts.Period, state.cfg.Contracts.RenewWindow)
+	hostInfos := make(map[types.PublicKey]cachedHostInfo)
+	for _, h := range hosts {
+		// ignore the pricetable's HostBlockHeight by setting it to our own blockheight
+		h.PriceTable.HostBlockHeight = state.cs.BlockHeight
+
+		isUsable, unusableResult := isUsableHost(state.cfg, state.rs, gc, f, h, minScore, storedData[h.PublicKey])
```

Comment:
I'm not sure I'm a fan of this. Now we calculate a host's usability in several places. Can't we either update the cache for all hosts after running contract checks, or use the cache in the contract checks? I feel it's a bit odd to have usability checks be performed completely separate from the checks where usability actually is taken into account and matters.

It offers room for error. The ignoring if the price table's host block height for instance is a good example. We might remove it in the checks, or update it here or whatever, that would result in the contractor marking a host as not usable and the API returning `UnusableResult` that don't match the decision making.

Reply:
Initially I wanted to reuse the cache but that's a bit annoying. We already compute the usability in multiple places for different reasons.
One time we compute it before we can even build the cache. That's because we need to compute it to determine a min score which we need to build the cache.
Another time we don't look at all hosts but only the ones with contracts and use fresh price tables.
One time we can actually use the cache instead but it's within `candidateHosts` which is called twice atm. So we'd need to refactor that to once be called with cached hosts and once with fresh values so we don't really reduce the number of times we compute it after all.
Doing it in the contract checks is the place that doesn't work since we only look at hosts with contracts and not all hosts.